algos/src/sample_rate_conversion.rst

```diff
@@ -349,94 +343,92 @@ the output is:
 		Arg:         -d -r 32000 -R 48000 -i chirp_test_in.raw -o chirp_test_out.raw -t ../../test/topology/test-playback-ssp2-mclk-0-I2S-src-s32le-s32le-48k-24576k-nocodec.tplg -a src=libsof_src.so -b S32_LE
 		Ld lib path: ../../testbench/build_testbench/sof_ep/install/lib:../../testbench/build_testbench/sof_parser/install/lib
 
-When debugging the testbench library path need to be appended to
-environment variable LD_LIBRARY_PATH and the shown arguments need to
+When debugging the testbench, the library path needs to be appended to
+the environment variable LD_LIBRARY_PATH, and the shown arguments need to
 be set for the debugger such as text mode gdb or graphical ddd. If the
 option to not delete audio data files the test input files can be used
 for debugging as well.
 
-Currently the testbench can be debugged only as host
-(x86) gcc build but debugging possibility with xt-gdb will be restored
-to be able to debug also xtensa optimized version of component in
-testbench.
-   
+Currently the testbench can be debugged only as the host (x86) gcc build but
+debugging possibility with xt-gdb will be restored to be able to debug also
+xtensa optimized version of component in the testbench.
+
 Polyphase decomposition
 ***********************
 
 The SRC component implementation is based on this algorithm level
-optimization. The next figure shows derivation of polyphase fractional
-resampler for 3/4 ratio that is used in e.g. 32 to 24 kHz conversion.
+optimization. The next figure shows derivation of the polyphase fractional
+resampler for a 3/4 ratio that is used in , for example, a 32 to 24 kHz conversion.
 
 .. figure:: poly34.png
 
    Polyphase decomposition for fractional 32 to 24 kHz conversion (3/4)
 
-1. the basic conversion is shown. 
+1. The basic conversion is shown.
 
 2. The interpolation is changed to polyphase filter where low-pass
-   filter H(z) is split to three sub-filters R\ :sub:`0`\(z),
+   filter H(z) is split into three sub-filters R\ :sub:`0`\(z),
    R\ :sub:`1`\(z), and R\ :sub:`2`\(z).
 
-3. The “3 to 1 commutator” structure the z\ :sup:`-1` unit delays are
+3. The “3 to 1 commutator” structure that the z\ :sup:`-1` unit delays are
    multiplicated to match the decimation rate of 4. The subfilter outputs
    need to be compensated with additional negative delay (zp, p > 0)
-   added to preserve the sub-filter out to whole filter chain output
-   Y(z).
+   to preserve the sub-filter out to whole filter chain output Y(z).
 
-4. The added negative delays are moved to filter input side by
-   dividing the negative delay by interpolation factor. Also the
-   decimation at filter output is moved to commutator input side.
+4. The added negative delays are moved to filter the input side by
```

Comment:
This change is confusing: "The added negative delays are moved to filter the input side". In the picture it's clearly seen that we moved the negative delay elements from output (right side) towards input (left size).

Reply:
@singalsu I only inserted the words "the" in your original sentence. What change is confusing?

Reply:
Maybe it made my grammar mistake to stand out. Something like "The added negative delays are moved to the input side of the filter" is better?

Reply:
@singalsu Yes! That is better and completely understandable. Do you mind making this update when you add the new diagrams? Thanks!

Reply:
Yep, I'll include this.